diff -urN epic4-1.0.1.org/configure.in epic4-1.0.1/configure.in
--- epic4-1.0.1.org/configure.in	Sun Dec 30 18:16:10 2001
+++ epic4-1.0.1/configure.in	Sun Dec 30 18:16:29 2001
@@ -218,8 +218,8 @@
 dnl originally wanted to have it around, but svr3 insisted not so. (grrr)
 dnl
 
-AC_CHECK_FUNC(getpgid, AC_DEFINE(HAVE_GETPGID),)
-AC_CHECK_FUNC(getpgrp, AC_FUNC_GETPGRP,)
+AC_CHECK_FUNC(getpgid, [AC_DEFINE(HAVE_GETPGID)],)
+AC_CHECK_FUNC(getpgrp, [AC_FUNC_GETPGRP],)
 
 
 dnl ----------------------------------------------------------
