--- epic4-1.1.7/source/server.c.orig	2003-07-20 09:47:35.000000000 +0200
+++ epic4-1.1.7/source/server.c	2003-07-20 09:46:57.000000000 +0200
@@ -2172,8 +2172,8 @@
 
 	if (!(host = strchr(userhost, '@')))
 	{
-		yell("Cannot set your userhost to [%s] because it does not
-		      contain a @ character!", userhost);
+		yell("Cannot set your userhost to [%s] because it does not "
+		     "contain a @ character!", userhost);
 		return;
 	}
 
