--- epic4-2.10.6/configure.in~	2022-08-26 09:31:02.000000000 +0200
+++ epic4-2.10.6/configure.in	2022-08-26 09:34:48.375184978 +0200
@@ -639,20 +639,12 @@
 [  --with-perl[=PATH]       Compile with perl support.],
 [ test -z "$withval" && withval=yes ],
 )
-test "xno" != "x$withval" && for base in $withval /usr/local /usr /
-do
-	FOUND=
-	for ext in so a dll ; do
-		test -r $base/lib/libperl.$ext && FOUND=1
-		test -r $base/libperl.$ext && FOUND=1
-	done
-	test -n "$FOUND" || continue
+if test "xno" != "x$withval"; then
 	AC_MSG_RESULT(yes)
 	LIBS="$LIBS `perl -MExtUtils::Embed -e ldopts`"
 	PERLDOTOH="perl.o"
 	AC_DEFINE(PERL)
-	break
-done
+fi
 test -z "$PERLDOTOH" && AC_MSG_RESULT(no)
 withval=''
 
