--- epic4-2.10.6/configure.in.orig	2024-07-01 14:09:38.511727569 +0200
+++ epic4-2.10.6/configure.in	2024-07-01 16:41:45.537845474 +0200
@@ -521,8 +521,10 @@
 ],[AC_MSG_RESULT(yes)])
 
 AC_MSG_CHECKING(for struct sockaddr_storage)
-AC_EGREP_HEADER([struct( |	)*sockaddr_storage], sys/socket.h, 
-  AC_MSG_RESULT(yes) 
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+    [#include <sys/socket.h>],
+    [struct sockaddr_storage ss;])],
+  AC_MSG_RESULT(yes)
   AC_DEFINE(HAVE_STRUCT_SOCKADDR_STORAGE),
   AC_MSG_RESULT(no))
 
